arch/arm/crc32_acle.c

```diff
@@ -17,34 +17,40 @@ uint32_t crc32_acle(uint32_t crc, const uint8_t *buf, size_t len) {
     Z_REGISTER uint32_t c;
     Z_REGISTER const uint16_t *buf2;
     Z_REGISTER const uint32_t *buf4;
+    Z_REGISTER const uint64_t *buf8;
 
     c = ~crc;
-    if (len && ((ptrdiff_t)buf & 1)) {
-        c = __crc32b(c, *buf++);
-        len--;
-    }
-
-    if ((len >= sizeof(uint16_t)) && ((ptrdiff_t)buf & sizeof(uint16_t))) {
-        buf2 = (const uint16_t *) buf;
-        c = __crc32h(c, *buf2++);
-        len -= sizeof(uint16_t);
-        buf4 = (const uint32_t *) buf2;
-    } else {
-        buf4 = (const uint32_t *) buf;
-    }
 
-#if defined(__aarch64__) || defined(_M_ARM64)
```

Comment:
Is it okay that this #ifdef was removed?

Reply:
The issue isn't necessary removing it, but that the code below for AArch64 also changes, which is unacceptable.

Reply:
Some compilers might not implement the 64-bit wide intrinsics for 32-bit ARM, but that's what for we test 32-bit ARM with both Windows and Linux compilers.

Reply:
Instead of checking if the architecture is 32-bit or 64-bit, we should only check that we're not using `_M_ARM` as that definitely doesn't support `__crc32d` and is likely to not support when ARMv8 is added as valid target.